packages/toc/src/generators/notebook/get_code_cell_heading.ts

```diff
@@ -22,14 +22,16 @@ type onClickFactory = (line: number) => () => void;
  * @param executionCount - execution count
  * @param lastLevel - last heading level
  * @param cellRef - cell reference
+ * @param index - index of referenced cell relative to other cells in the notebook
  * @returns notebook heading
  */
 function getCodeCellHeading(
   text: string,
   onClick: onClickFactory,
   executionCount: string,
   lastLevel: number,
-  cellRef: Cell
+  cellRef: Cell,
+  index: number
```

Comment:
In order to keep this logic backward-compatible until the next major version, I would suggest setting a default value, perhaps `index = -1` so that in the body of the function you can use the old logic if `!~index` and the new logic otherwise. You can add a deprecation warning that the `index` argument of the function will become mandatory in the next version. That way this PR will not be blocked pending a major release.

Reply:
@afshin, thanks for the review! I just pushed a new commit with those changes. Let me know if I missed something there.